.github/workflows/build-test-smart-contracts.yaml

```diff
@@ -187,19 +161,14 @@ jobs:
           - 'fib'
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - name: Install Rust
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: ${{ env.RUST_CLIPPY }}
-          override: true
-          target: wasm32-unknown-unknown
-          components: clippy
+        uses: actions/checkout@v4
       - name: Clippy
         working-directory: smart-contracts/rust-contracts/example-contracts/${{ matrix.example-contract }}
         run: |
           git config --global url."https://github.com/".insteadOf "git@github.com:"
+          rustup default ${{ env.RUST_CLIPPY }}
+          rustup component add clippy
+          rustup target add wasm32-unknown-unknown
```

Comment:
Do we need the `target` here?

Reply:
Yes, because some example contracts have a `.cargo/config.toml` which specifies the wasm target.